### PR TITLE
Create autoyast installation without registration for minimal update qcow on SLES12SP5

### PR DESCRIPTION
**Pull request by coolgw**
Create autoyast installation without registration for minimal update qcow on SLES12SP5

- Related ticket: https://progress.opensuse.org/issues/130489
- Needles: na
- Verification run: 
      Publish qcow: https://openqa.suse.de/tests/11437895
      Migration: https://openqa.suse.de/tests/11437916

Gitlab new case MR: https://gitlab.suse.de/qa-maintenance/qam-openqa-yml/-/merge_requests/555

### CODE REVIEW

File: data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml [hunk 1]
```diff
@@ -0,0 +1,1661 @@
...
+      <trusted_grub>false</trusted_grub>
+      <xen_append>crashkernel=173M,high</xen_append>
+      <xen_kernel_append>vga=gfx-1024x768x16 crashkernel=173M\&lt;4G</xen_kernel_append>
+    </global>
+    <loader_type>grub2</loader_type>
+  </bootloader>
+  <deploy_image>
+    <image_installation config:type="boolean">false</image_installation>
+  </deploy_image>
+  <firewall>
```
Comment (lemon-suse) [inline data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml:31]: I think we can remove the section for firewall.

File: data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml [hunk 2]
```diff
@@ -0,0 +1,1661 @@
+<?xml version="1.0"?>
+<!DOCTYPE profile>
+<profile xmlns="http://www.suse.com/1.0/yast2ns" xmlns:config="http://www.suse.com/1.0/configns">
+  <add-on>
+    <add_on_products config:type="list"/>
+  </add-on>
+  <bootloader>
```
Comment (lemon-suse) [inline data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml:7]: We can compact this part refer to https://openqa.nue.suse.com/tests/11164010/settings/autoyast_sle15/mini.xml

Ex:
<bootloader>
        <global>
            <timeout config:type="integer">-1</timeout>
        </global>
    </bootloader>

File: data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml [hunk 3]
```diff
@@ -0,0 +1,1661 @@
...
+      <package>sles-release</package>
+      <package>numactl</package>
+      <package>irqbalance</package>
+      <package>grub2</package>
+      <package>glibc</package>
+      <package>e2fsprogs</package>
+      <package>btrfsprogs</package>
+      <package>SuSEfirewall2</package>
+    </packages>
+    <patterns config:type="list">
```
Comment (lemon-suse) [inline data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml:1065]: This part looks like default patterns while not all patterns.

File: data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml [hunk 4]
```diff
@@ -0,0 +1,1661 @@
+<?xml version="1.0"?>
```
Comment (lemon-suse) [inline data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml:1]: It is too large profile, normally we have the final profile less then 500 lines. :) So need compact.

File: data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml (hunk 3 above)
Comment (coolgw) [reply data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml:396]: add all patterns

File: data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml (hunk 4 above)
Comment (coolgw) [reply data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml:1]: compact version done

File: data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml (hunk 1 above)
Comment (coolgw) [reply data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml:31]: remove rules parts, just keep firewall disable status make sure not impact case following.

File: data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml [hunk 5]
```diff
@@ -0,0 +1,505 @@
...
```
Comment (lemon-suse) [inline data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml:1]: For the profile name, it's better to s/autoinst/autoyast/. Others LGTM.

Comment (lemon-suse) [reply data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml:1]: We need reflect all patterns in name, so data/autoyast_sle12/autoyast_sles12sp5_media_all_patterns_minimal_x86_64.xml WDYT?

File: data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml (hunk 3 above)
Comment (chcao) [reply data/autoyast_sle12/autoinst_sles12sp5_media_minimal_x86.xml:396]: The matrix  is updated to use default patterns for minimal update cases, so need to add the default pattern list in the ay profile.

**Comment by rakoenig** [comment]
Thinking of the 2 tickets that say "move autoyast profiels to data/yam" I wonder if we should start doing this here as well...

**Comment by jknphy** [comment]
@coolgw we changed from all patterns to default patterns in the middle of your development, is still fine for you to make it default or do you prefer for this to be merged and create a follow-up ticket to adjust the patterns?

**Comment by coolgw** [comment]
> @coolgw we changed from all patterns to default patterns in the middle of your development, is still fine for you to make it default or do you prefer for this to be merged and create a follow-up ticket to adjust the patterns?

Current PR contain both def and all patterns which can satifiy your requirment. We do not need follow-up ticket

**Pull request merged**